config/jobs/kubernetes/sig-testing/apidiff.yaml

```diff
@@ -56,5 +56,82 @@ presubmits:
             cpu: 2
             memory: 12Gi
 
-# A periodic job which shows API diffs for staging repos since the last release
-# might be useful. Not done yet.
+  - name: pull-kubernetes-apidiff-client-go
+    cluster: eks-prow-build-cluster
+    # A job which automatically runs for changes in client-go or the generated code
+    # to have visibility on the changes that will impact the external projects
+    # that are using the Kubernetes golang clients
+    run_if_changed: '(^staging\/src\/k8s.io\/client-go)|(^staging\/src\/k8s.io\/code-generator\/examples)'
+    optional: true
+    decorate: true
+    annotations:
+      # The apidiff.sh script uses the latest revision of apidiff.
+      # There is no guarantee that this will continue to work for
+      # older branches, so let's not even create per-release
+      # copies of this job.
+      fork-per-release: "false"
+      testgrid-dashboards: sig-testing-misc
+      testgrid-create-test-group: 'true'
+    path_alias: k8s.io/kubernetes
+    spec:
+      containers:
+      - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20241128-8df65c072f-master
+        command:
+        - runner.sh
+        args:
+        - /bin/sh
+        - -c
+        # PULL_BASE_SHA is the revision on the target branch that the
+        # PR gets merged into. What we want instead is the revision at
+        # which the PR branch diverged from the target branch.
+        # We get that from "git merge-base".
+        - "./hack/apidiff.sh -r $(git merge-base ${PULL_BASE_SHA} ${PULL_PULL_SHA}) -t ${PULL_PULL_SHA} ./staging/src/k8s.io/code-generator/examples ./staging/src/k8s.io/client-go"
+        env:
+        - name: REPO_DIR
+          value: /workspace/k8s.io/kubernetes
+        resources:
+          # Memory limits are derived from pull-kubernetes-verify, with less CPUs.
+          limits:
+            cpu: 2
+            memory: 12Gi
+          requests:
+            cpu: 2
+            memory: 12Gi
+
+
+periodics:
+# A periodic job which shows API diffs for staging repos per day
+- name: ci-kubernetes-apidiff-daily
+  cluster: eks-prow-build-cluster
+  interval: 24h
```

Comment:
this approach isn't technically guaranteed to schedule daily, you should probably use a cron for this and pick a time of day to run for consistency

Reply:
it's a canary job, I just want to get some periodic signal , nothing else, if this work well we figure out a more elegant way of scheduling, but in the meantime think on it as a personal job, nothing else

Reply:
OK, it's not that complicated though ... https://docs.prow.k8s.io/docs/jobs/#how-to-configure-new-jobs

You just swap this field for `cron: <normal cron string>`, e.g. `cron: 0 0 * * *` would run every day at midnight UTC

Reply:
If the time window matches, the report will be more useful, otherwise it will vary by how many developers have recently been merging code. not blocking, but consider this in a follow-up. We can look at the merge graph in k/k.

Reply:
this thread is **non-blocking**, we can do this in a follow-up, but I strongly recommend it.

`interval: 24h` is almost always not what people actually want, `interval` makes sense for frequently run things where we don't want to stack up missed runs from the job potentially taking longer than the interval, and we don't care at all about timing otherwise. 

pretty much any job with a long interval is using a cron instead for this reason, the scheduling is more predictable.

for a daily report you want predictable, so you can reliably check it after it's done. 
the timing of interval is affected by when it was first introduced, any disruptions to prow, and any other jitter, since it's always computed since the last run.